src/mococrw/hash.h

```diff
@@ -0,0 +1,54 @@
+/*
+ * #%L
+ * %%
+ * Copyright (C) 2018 BMW Car IT GmbH
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+#pragma once
+
+#include "mococrw/openssl_wrap.h"
+
+namespace mococrw
+{
+
+void sha256(std::string message, std::string *outDigest);
+void sha256(std::vector<uint8_t> message, std::vector<uint8_t> *outDigest);
+void sha256(uint8_t* message, int messageLength, std::vector<uint8_t> *outDigest, int *outDigestLength);
+
+void sha512(std::string message, std::string *outDigest);
+void sha512(std::vector<uint8_t> message, std::vector<uint8_t> *outDigest);
+void sha512(uint8_t* message, int messageLength, std::vector<uint8_t> *outDigest, int *outDigestLength);
+
+class Hash
+{
+public:
+    static Hash sha256();
+    static Hash sha512();
+    void digest (std::vector<uint8_t> &outValue);
```

Comment:
see above: Why not just return it? If we want to offer a function to return hex here, we can also add (like it is in the python api) a hexdigest function.

Reply:
all functions return vectors now. hex can be reached by using utility::hex() so hexdigest ist probably not needed